include/oneapi/dpl/internal/scan_by_segment_impl.h

```diff
@@ -0,0 +1,372 @@
+                                {
+                                    __local_collector = __binary_op(__partials_acc[__j], __local_collector);
+                                    if (__seg_ends_acc[__j] || __j == 0)
+                                    {
+                                        __loc_seg_ends_acc[__local_id] = true;
+                                        break;
+                                    }
+                                }
+                                __loc_partials_acc[__local_id] = __local_collector;
+                                __dpl_sycl::__group_barrier(__item);
+                                // serial aggregate collection and synchronization
+                                if (__local_id == 0)
+                                {
+                                    for (int64_t __j = 0; __j < __wgroup_size; ++__j)
+                                    {
+                                        __agg_collector = __binary_op(__loc_partials_acc[__j], __agg_collector);
+                                        if (__loc_seg_ends_acc[__j])
+                                        {
+                                            __last_it = true;
+                                            break;
+                                        }
+                                    }
+                                }
+                                __agg_collector = __dpl_sycl::__group_broadcast(__item.get_group(), __agg_collector);
+                                __last_it = __dpl_sycl::__group_broadcast(__item.get_group(), __last_it);
+                                if (__last_it)
+                                    break;
+                            }
+                            __ag_exists = __dpl_sycl::__any_of_group(__group, __ag_exists);
+
+                            if (!__ag_exists)
+                                return;
+
+                            // 2c. Second pass over the keys, reidentifying end segments and applying work group
+                            // aggregates if appropriate.
+
+                            ::std::size_t __end_nm1_cap =
+                                __dpl_sycl::__minimum<decltype(__n)>{}(__start + __vals_per_item, __n - 1);
+                            ::std::size_t __local_min_key_idx = __n - 1;
+
+                            // Find the smallest index in the work group
+                            for (::std::size_t __i = __end_nm1_cap - 1; __i >= __start; --__i)
+                            {
+                                if (!__binary_pred(__keys[__i], __keys[__i + 1]))
+                                {
+                                    __local_min_key_idx = __i;
+                                }
+                            }
+
+                            ::std::size_t __wg_min_seg_end = __dpl_sycl::__reduce_over_group(
+                                __group, __local_min_key_idx, __dpl_sycl::__minimum<decltype(__local_min_key_idx)>());
```

Comment:
I think you can replace the `decltype` with `::std::size_t`, since you declared that variable just a few lines above.

Reply:
Will make this change.